src/CADL.Extension/Emitter.Csharp/src/lib/cadlServer.ts

```diff
@@ -38,18 +38,19 @@ export function resolveServers(
     return servers.map((server) => {
         const parameters: InputParameter[] = [];
         let url: string = server.url;
+        let endpoint: string = url.replace("http://", "").replace("https://", "").split("/")[0];
         for (const [name, prop] of server.parameters) {
             // if (!validateValidServerVariable(program, prop)) {
             //   continue;
             // }
-
+            const isEndpoint: boolean = endpoint === (`{${name}}`);
```

Comment:
We probably can be even more specific and check for name `"$host"`, similar to M4: https://github.com/Azure/autorest/blob/5149dd0311cd5fad437ac18cff1e387b0835c1a5/packages/extensions/modelerfour/src/modeler/modelerfour.ts#L1748

Reply:
> We probably can be even more specific and check for name `"$host"`, similar to M4: https://github.com/Azure/autorest/blob/5149dd0311cd5fad437ac18cff1e387b0835c1a5/packages/extensions/modelerfour/src/modeler/modelerfour.ts#L1748

Hello @AlexanderSher cadl will not define host parameter as global client parameter, it only define `@server` to define url. So we may not need to check this.